module/VuFind/src/VuFind/Favorites/FavoritesService.php

```diff
@@ -368,6 +428,24 @@ public function saveListForUser(UserListEntityInterface $list, ?UserEntityInterf
         $this->rememberLastUsedList($list);
     }
 
+    /**
+     * Add a tag to a list.
+     *
+     * @param string                  $tagText The tag to save.
+     * @param UserListEntityInterface $list    The list being tagged.
+     * @param UserEntityInterface     $user    The user posting the tag.
+     *
+     * @return void
+     */
+    public function addListTag(string $tagText, UserListEntityInterface $list, UserEntityInterface $user): void
+    {
+        $tagText = trim($tagText);
+        if (!empty($tagText)) {
+            $tag = $this->getDbTable('tags')->getByText($tagText);
```

Comment:
Database service code in a non-database service?

Reply:
Same here. I will address this soon, but in a separate PR.